core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -250,7 +250,7 @@ public Object run() throws Exception {
                 String errorMessage = e.getCause().getCause().getMessage();
                 if (isClientException) {
                     if (StringUtils.isBlank(errorCode)) {
-                        throw new UserStoreClientException(errorCode, e);
+                        throw new UserStoreClientException(errorMessage, e);
```

Comment:
We validate based on the error message in the other components. So if the error message is null, it is handled by using the general error message. Checked the other places as well.